### PR TITLE
avoid stackoverflow in x509 http header mode - port forward

### PR DESCRIPTION
**Pull request by hdeadman**
Same as: https://github.com/apereo/cas/pull/5281

### CODE REVIEW

**Comment by hdeadman** [comment]
@apereocas-bot runci

File: support/cas-server-support-x509-webflow/src/main/java/org/apereo/cas/web/flow/X509CertificateCredentialsRequestHeaderAction.java [hunk 1]
```diff
@@ -54,4 +59,10 @@ protected Credential constructCredentialsFromRequest(final RequestContext contex
         }
         return null;
     }
+
+    @Override
+    protected void onError(final RequestContext requestContext) {
+        WebUtils.putCasLoginFormViewable(requestContext, casProperties.getAuthn().getX509().isMixedMode());
```
Comment (mmoayyed) [inline support/cas-server-support-x509-webflow/src/main/java/org/apereo/cas/web/flow/X509CertificateCredentialsRequestHeaderAction.java:65]: You might need to get the existing value and OR it with the setting here.

File: support/cas-server-support-x509-webflow/src/main/java/org/apereo/cas/web/flow/X509CertificateCredentialsNonInteractiveAction.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
      */
     public static final String REQUEST_ATTRIBUTE_X509_CERTIFICATE = "javax.servlet.request.X509Certificate";
     
-    private static final String REQUEST_ATTRIBUTE_X509_ERROR = "X509CertificateAuthenticationError";
+    public static final String REQUEST_ATTRIBUTE_X509_ERROR = "X509CertificateAuthenticationError";
```
Comment (mmoayyed) [inline support/cas-server-support-x509-webflow/src/main/java/org/apereo/cas/web/flow/X509CertificateCredentialsNonInteractiveAction.java:34]: I suspect the build fails with a "public field needs javadoc".

**Review by mmoayyed — CHANGES_REQUESTED** [review]
Looks great.

File: support/cas-server-support-x509-webflow/src/main/java/org/apereo/cas/web/flow/X509CertificateCredentialsRequestHeaderAction.java (hunk 1 above)
Comment (hdeadman) [reply support/cas-server-support-x509-webflow/src/main/java/org/apereo/cas/web/flow/X509CertificateCredentialsRequestHeaderAction.java:65]: Should that be done for the both X509 credential actions?

Comment (mmoayyed) [reply support/cas-server-support-x509-webflow/src/main/java/org/apereo/cas/web/flow/X509CertificateCredentialsRequestHeaderAction.java]: I'd say so, yes. It's an oversight if they are not doing that already.

Comment (hdeadman) [reply support/cas-server-support-x509-webflow/src/main/java/org/apereo/cas/web/flow/X509CertificateCredentialsRequestHeaderAction.java]: I added a new method to WebUtils b/c `isCasLoginFormViewable` was defaulting to TRUE, which is good for use in UI, but not in OR statement.

**Pull request merged**